amazon-kinesis-client/src/test/java/software/amazon/kinesis/coordinator/DiagnosticEventsTest.java

```diff
@@ -0,0 +1,152 @@
+import software.amazon.kinesis.leases.LeaseCoordinator;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadPoolExecutor;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@Slf4j
+@RunWith(MockitoJUnitRunner.class)
+public class DiagnosticEventsTest {
+    @Mock
+    private ThreadPoolExecutor executor;
+    @Mock
+    private LeaseCoordinator leaseCoordinator;
+    @Mock
+    private DiagnosticEventHandler defaultHandler;
+
+    private DiagnosticEventHandler customHandler;
+    private boolean wasCustomHandlerInvoked;
+
+    private Throwable throwable;
+
+    private int activeThreadCount;
+    private int corePoolSize;
+    private int largestPoolSize;
+    private int maximumPoolSize;
+
+    private SynchronousQueue<Runnable> executorQueue;
+    private Collection<Lease> leaseAssignments;
+
+    @Before
+    public final void setup() {
+        customHandler = new CustomHandler();
+        wasCustomHandlerInvoked = false;
+
+        throwable = new TestRejectedTaskException();
+
+        activeThreadCount = 2;
+        corePoolSize = 4;
+        largestPoolSize = 8;
+        maximumPoolSize = 16;
+
+        executorQueue = new SynchronousQueue<>();
+
+        final Lease lease = new LeaseBuilder().build();
+        leaseAssignments = Collections.singletonList(lease);
+
+        when(executor.getQueue()).thenReturn(executorQueue);
+        when(executor.getActiveCount()).thenReturn(activeThreadCount);
+        when(executor.getCorePoolSize()).thenReturn(corePoolSize);
+        when(executor.getLargestPoolSize()).thenReturn(largestPoolSize);
+        when(executor.getMaximumPoolSize()).thenReturn(maximumPoolSize);
+        when(leaseCoordinator.getAssignments()).thenReturn(leaseAssignments);
+    }
+
+    @Test
+    public final void testExecutorStateEventWithDefaultHandler() {
+        ExecutorStateEvent event = new ExecutorStateEvent(executor, leaseCoordinator);
+        event.accept(defaultHandler);
+
+        assertEquals(event.getActiveThreads(), activeThreadCount);
+        assertEquals(event.getCoreThreads(), corePoolSize);
+        assertEquals(event.getLargestPoolSize(), largestPoolSize);
+        assertEquals(event.getMaximumPoolSize(), maximumPoolSize);
+        assertEquals(event.getLeasesOwned(), leaseAssignments.size());
+        assertEquals(event.getCurrentQueueSize(),0);
+
+        verify(defaultHandler, times(1)).visit(event);
+    }
+
+    @Test
+    public final void testExecutorStateEventWithCustomHandler() {
+        ExecutorStateEvent event = new ExecutorStateEvent(executor, leaseCoordinator);
+        event.accept(customHandler);
+
+        assertTrue(wasCustomHandlerInvoked);
+        wasCustomHandlerInvoked = false;
```

Comment:
Not sure why this statement is the last statement. It doesn't seem to do anything.

Reply:
This resets wasCustomHandlerInvoked so that it doesn't start out as true on subsequent tests and give false positives. Will add a comment for this

Reply:
The `@Before` method gets run before every test. The value is being set as false in the before.

Reply:
You're right, redundant line. Removed it from both tests.